### PR TITLE
Header and version check in log-parse command

### PR DESCRIPTION
**Pull request by xyan264**
**Background:** 
  The previous #266 adds a header to the binary log file generated from `log-dump` command. The header includes signature and version information.

**This PREP:**
- Updates `switchtec_parse_log` function to handle the header that _might_ exist in the input log file 
- Adds version checking to `log-parse` command, with a print to remind user when the input log file and log def file versions don't match

### CODE REVIEW

**Review by lsgunth — APPROVED** [review]
Looks good to me.

File: cli/main.c [hunk 1]
```diff
@@ -1097,6 +1097,10 @@ static int log_parse(int argc, char **argv)
 		fprintf(stderr, "\nParsed log saved to %s.\n",
 			cfg.parsed_log_filename);
 
+	if (ret == ENOEXEC)
+		fprintf(stderr, "\nWARNING: The two input files have different version numbers.\n"
+				"The log file is parsed but the output file might contain errors.\n");
+
```
Comment (kelvin-cao) [inline cli/main.c:1103]: It can be helpful to print the versions that mismatch.

Comment (xyan264) [reply cli/main.c:1103]: At this level there is no function to get the version numbers of log file and log-def file. 

Do you want me to add 2 functions like:
```
int switchtec_log_get_header(struct log_header *header)    //returns bin header info
int switchtec_log_def_get_header(struct log_def_header *header) //returns log-def header info
```
so we can print the versions from `header`?

I don't see many use cases for these 2 functions, but it doesn't hurt to add these.

Comment (kelvin-cao) [reply cli/main.c:1103]: How about add an out structure parameter to switchtec_parse_log() to pass out the versions of log and def files?

Comment (xyan264) [reply cli/main.c:1110]: Good idea. Code is updated. Output looks like this:
```./switchtec log-parse switchtec-ram.log oslog-def-445.dat

Parsed log saved to log.txt.

WARNING: The two input files have different version numbers.
                FW Version      SDK Version
Log file:       0x06010d0e      0x00000000
Log def file:   0x06010d0e      0x06010d0e

The log file is parsed but the output file might contain errors.```

**Review by kelvin-cao — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**